connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java

```diff
@@ -0,0 +1,242 @@
+import javax.ws.rs.core.Response;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(Enclosed.class)
+public class RestClientTest {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<TestDTO> TEST_TYPE = new TypeReference<TestDTO>() {
+    };
+    private static final SecretKey MOCK_SECRET_KEY = getMockSecretKey();
+
+    private static void assertIsInternalServerError(ConnectRestException e) {
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), e.statusCode());
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), e.errorCode());
+    }
+
+    private static SecretKey getMockSecretKey() {
+        SecretKey mockKey = mock(SecretKey.class);
+        when(mockKey.getFormat()).thenReturn("RAW");// supported format by
+        when(mockKey.getEncoded()).thenReturn("SomeKey".getBytes(StandardCharsets.UTF_8));
+        return mockKey;
+    }
+
+    private static RestClient.HttpResponse<TestDTO> httpRequest(HttpClient httpClient, String requestSignatureAlgorithm) {
+        return RestClient.httpRequest(
+                httpClient,
+                "https://localhost:1234/api/endpoint",
+                "GET",
+                null,
+                new TestDTO("requestBodyData"),
+                TEST_TYPE,
+                MOCK_SECRET_KEY,
+                requestSignatureAlgorithm);
+    }
+
+    private static RestClient.HttpResponse<TestDTO> httpRequest(HttpClient httpClient) {
+        String validRequestSignatureAlgorithm = "HmacSHA1";
+        return httpRequest(httpClient, validRequestSignatureAlgorithm);
+    }
+
+
+    @RunWith(Parameterized.class)
+    public static class RequestFailureParameterizedTest {
+        private final HttpClient httpClient = mock(HttpClient.class);
+
+        @Parameterized.Parameter
+        public Throwable requestException;
+
+        @Parameterized.Parameters
+        public static Collection<Object[]> requestExceptions() {
+            return Arrays.asList(new Object[][]{
+                    {new InterruptedException()},
+                    {new ExecutionException(null)},
+                    {new TimeoutException()}
+            });
+        }
+
+        private static Request buildThrowingMockRequest(Throwable t) throws ExecutionException, InterruptedException, TimeoutException {
+            Request req = mock(Request.class);
+            when(req.send()).thenThrow(t);
```

Comment:
Because we don't set up any expectations for `Request::header`, we get a `NullPointerException` right now at this line during these tests, instead of the exception we want to be thrown from `Request::send`: https://github.com/apache/kafka/blob/6495a0768cae1086d4b1dfa466967dfe178c1553/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/InternalRequestSignature.java#L57-L58

Luckily, there's an easy fix for this:
```suggestion
            when(req.send()).thenThrow(t);
            when(req.header(anyString(), anyString())).thenReturn(req);
```

Reply:
@C0urante I believe this is already in the latest commit, any chance you were referring to a the previous commit?

Reply:
nice catch, thx.  setupHttpClient does this but I forgot to do it here and didn't notice the NPEs.
fixed